server/src/main/java/org/elasticsearch/health/HealthIndicatorService.java

```diff
@@ -13,5 +13,13 @@
  */
 public interface HealthIndicatorService {
 
+    String name();
```

Comment:
I was considering enums, but then returned back to string. It might be impossible to introduce new names and components from other modules and plugins this way. As a middle ground we might introduce special types for name and component similar to `DiscoveryNodeRole`?